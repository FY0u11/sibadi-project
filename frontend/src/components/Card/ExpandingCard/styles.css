.my-expanding-card {
  margin: 0 auto 20px auto;
}

.my-expanded {
  border-bottom: 0;
  border-radius: 4px 4px 0 0;
}

.my-expanded-body {
  border-left: 1px solid #dfdfdf;
  border-right: 1px solid #dfdfdf;
  border-bottom: 1px solid #dfdfdf;
  border-radius: 0 0 4px 4px;
  margin-bottom: 20px;
  padding: 20px;
  background: #f7f7f7;
}
