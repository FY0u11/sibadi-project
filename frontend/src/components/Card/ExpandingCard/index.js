import React from 'react'

import './styles.css'

const ExpandingCard = ({ image, title, body, exBody, isExpanded, width = '500px' }) => {
  return (
    <div className="my-expanding-card" style={{ width }}>
      <div className={`card ${isExpanded && 'my-expanded'}`}>
        <div className="row no-gutters">
          <div className="col-md-4">{image}</div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className="card-title">{title}</h5>
              {body}
            </div>
          </div>
        </div>
      </div>
      {isExpanded && <div className="my-expanded-body">{exBody}</div>}
    </div>
  )
}

export default ExpandingCard
