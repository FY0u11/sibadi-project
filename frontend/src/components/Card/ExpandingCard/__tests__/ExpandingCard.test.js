import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import ExpandingCard from '../'

afterEach(cleanup)

describe('Testing ExpandingCard component', () => {
  it('It should be properly rendered in expanded form', () => {
    const props = {
      image: 'test image',
      title: 'test title',
      body: 'test body',
      exBody: 'expanded body',
      isExpanded: true
    }
    const { asFragment } = render(<ExpandingCard {...props} />)
    expect(asFragment()).toMatchSnapshot('1')
  })
})
