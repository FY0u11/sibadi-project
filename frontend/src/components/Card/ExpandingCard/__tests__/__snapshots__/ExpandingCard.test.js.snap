// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ExpandingCard component It should be properly rendered in expanded form: 1 1`] = `
<DocumentFragment>
  <div
    class="my-expanding-card"
    style="width: 500px;"
  >
    <div
      class="card my-expanded"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="col-md-4"
        >
          test image
        </div>
        <div
          class="col-md-8"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              test title
            </h5>
            test body
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-expanded-body"
    >
      expanded body
    </div>
  </div>
</DocumentFragment>
`;
